src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPEditorService.cs

```diff
@@ -223,6 +225,39 @@ private static void ApplyTextEdits(IEnumerable<TextEdit> textEdits, ITextSnapsho
             vsTextEdit.Apply();
         }
 
+        public override void MoveCaretToPosition(string fullPath, int absoluteIndex)
+        {
+            if (string.IsNullOrEmpty(fullPath))
+            {
+                throw new ArgumentNullException(nameof(fullPath));
+            }
+
+            VsShellUtilities.OpenDocument(_serviceProvider, fullPath, VSConstants.LOGVIEWID.TextView_guid, out _, out _, out var windowFrame);
```

Comment:
A lot of this is duplicated in `ApplyTextEditsAsync` were you thinking about keeping the logic isolated to make for easy removal later?

Reply:
Yup. I plan to remove this entire `MoveCaretToPosition` abstract method and its implementation once snippet support is in.